tfhe/src/core_crypto/algorithms/test/lwe_keyswitch_key_generation.rs

```diff
@@ -87,9 +87,13 @@ fn test_seeded_lwe_ksk_gen_equivalence<Scalar: UnsignedTorus>(
 
         assert!(check_content_respects_mod(&seeded_ksk, ciphertext_modulus));
 
-        let decompressed_ksk = seeded_ksk.decompress_into_lwe_keyswitch_key();
+        let ser_decompressed_ksk = seeded_ksk.clone().decompress_into_lwe_keyswitch_key();
```

Comment:
This is something that I/we alredy mentionned, but `decompress_into_lwe_ksk` should take `self` not `self`, that way we could avoid that useless clone

Reply:
I know we have an issue for it, but it still makes basically no sense to keep a compressed entity on hand so I'm still torn :/

Reply:
(this is a particular case for tests)

Reply:
also I like the fact once decompressed, and therefore when the compressed entity should not be useful anymore, it gets dropped

I can follow up with a PR for https://github.com/zama-ai/tfhe-rs-internal/issues/147 but I still feel like it's not super useful :/

Reply:
plus the algorithms allow to do just that, not consume the original entity, so yeah I'm fine with this design choice at the moment

Reply:
I could update the docstrings in a next PR to direct people to the algorithms that don't consume the entity ? wdyt ?

Reply:
lets leave it as is